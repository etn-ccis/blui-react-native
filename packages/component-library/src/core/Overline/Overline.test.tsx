import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Overline } from '.';

describe('Overline', () => {
    it('Overline Renders', () => {
        let renderer: TestRenderer.ReactTestRenderer;
        TestRenderer.act(() => {
            renderer = TestRenderer.create(<Overline>Overline</Overline>);
        });
        expect(renderer!.toJSON()).toMatchSnapshot();
    });

    it('Renders the color correctly', () => {
        let renderer: TestRenderer.ReactTestRenderer;
        TestRenderer.act(() => {
            renderer = TestRenderer.create(<Overline>Overline </Overline>);
        });
        expect(renderer!.toJSON()).toMatchSnapshot();
    });

    it('Accepts style override', () => {
        let renderer: TestRenderer.ReactTestRenderer;
        TestRenderer.act(() => {
            renderer = TestRenderer.create(<Overline style={{ color: 'blue', fontSize: 9 }}>Overline </Overline>);
        });
        expect(renderer!.toJSON()).toMatchSnapshot();
    });
});
