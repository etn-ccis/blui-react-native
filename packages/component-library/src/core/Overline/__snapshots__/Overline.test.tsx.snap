// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Overline Accepts style override 1`] = `
<Text
  allowFontScaling={true}
  maxFontSizeMultiplier={100}
  style={
    [
      {
        "textAlign": "left",
      },
      {
        "color": "rgba(28, 27, 31, 1)",
        "writingDirection": "ltr",
      },
      [
        {
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "500",
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        [
          {
            "color": "rgba(28, 27, 31, 1)",
            "letterSpacing": 2,
            "textTransform": "uppercase",
          },
          undefined,
          {
            "color": "blue",
            "fontSize": 9,
          },
        ],
      ],
    ]
  }
>
  Overline 
</Text>
`;

exports[`Overline Overline Renders 1`] = `
<Text
  allowFontScaling={true}
  maxFontSizeMultiplier={100}
  style={
    [
      {
        "textAlign": "left",
      },
      {
        "color": "rgba(28, 27, 31, 1)",
        "writingDirection": "ltr",
      },
      [
        {
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "500",
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        [
          {
            "color": "rgba(28, 27, 31, 1)",
            "letterSpacing": 2,
            "textTransform": "uppercase",
          },
          undefined,
          undefined,
        ],
      ],
    ]
  }
>
  Overline
</Text>
`;

exports[`Overline Renders the color correctly 1`] = `
<Text
  allowFontScaling={true}
  maxFontSizeMultiplier={100}
  style={
    [
      {
        "textAlign": "left",
      },
      {
        "color": "rgba(28, 27, 31, 1)",
        "writingDirection": "ltr",
      },
      [
        {
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "500",
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        [
          {
            "color": "rgba(28, 27, 31, 1)",
            "letterSpacing": 2,
            "textTransform": "uppercase",
          },
          undefined,
          undefined,
        ],
      ],
    ]
  }
>
  Overline 
</Text>
`;
